### PR TITLE
Def 8.28 and Prop 8.29

### LINKED ISSUE #17: Formalise two-sided discrete fibrations (Section 8.6 of RS17 paper)
- [ ] Definition 8.28
- [ ] Proposition 8.29

### PR DESCRIPTION
**Pull request by TashiWalde**
closes #17

### CODE REVIEW

File: src/simplicial-hott/08-covariant.rzk.md [hunk 1]
```diff
@@ -1565,3 +1565,28 @@ commuting with the contravariant lifts.
...
+  :=
+    product
+      ((a : A) -> is-covariant B (\b -> C a b))
+      ((b : B) -> is-contravariant A (\ a -> C a b))
+```
+
+```rzk title="RS17, Proposition 8.29"
+#def is-two-sided-discrete-representable-is-segal
+  (A : U)
+  (is-segal-A : is-segal A)
```
Comment (fredrik-bakke) [inline src/simplicial-hott/08-covariant.rzk.md:1585]:
```suggestion
  ( A : U)
  ( is-segal-A : is-segal A)
```

File: src/simplicial-hott/08-covariant.rzk.md [hunk 2]
```diff
@@ -1565,3 +1565,28 @@ commuting with the contravariant lifts.
     ( contravariant-fiberwise-transformation-application
         A x y f C D is-contravariant-C ϕ v)
 ```
+
+## Two sided discrete fibrations
+
+```rzk title="RS17, Definition 8.28"
+#def is-two-sided-discrete
+  (A B : U)
+  (C : A -> B -> U)
```
Comment (fredrik-bakke) [inline src/simplicial-hott/08-covariant.rzk.md:1574]:
```suggestion
  ( A B : U)
  ( C : A -> B -> U)
```

File: src/simplicial-hott/08-covariant.rzk.md [hunk 3]
```diff
@@ -1565,3 +1565,28 @@ commuting with the contravariant lifts.
...
+  : U
+  :=
+    product
+      ((a : A) -> is-covariant B (\b -> C a b))
+      ((b : B) -> is-contravariant A (\ a -> C a b))
```
Comment (fredrik-bakke) [inline src/simplicial-hott/08-covariant.rzk.md:1579]:
```suggestion
      ( (a : A) -> is-covariant B (\b -> C a b))
      ( (b : B) -> is-contravariant A (\ a -> C a b))
```

File: src/simplicial-hott/08-covariant.rzk.md [hunk 4]
```diff
@@ -1565,3 +1565,28 @@ commuting with the contravariant lifts.
...
+```
+
+```rzk title="RS17, Proposition 8.29"
+#def is-two-sided-discrete-representable-is-segal
```
Comment (fizruk) [inline src/simplicial-hott/08-covariant.rzk.md:1583]: I'm not sure "representable" should be part of the name here ("hom" seems more appropriate?).
@emilyriehl might know better what's the correct terminology here.

File: src/simplicial-hott/08-covariant.rzk.md [hunk 5]
```diff
@@ -1565,3 +1565,28 @@ commuting with the contravariant lifts.
...
+  (A : U)
+  (is-segal-A : is-segal A)
+  : is-two-sided-discrete A A (hom A)
+  :=
+    ( is-covariant-representable-is-segal A is-segal-A
+    , is-contravariant-representable-is-segal A is-segal-A
+    )
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/simplicial-hott/08-covariant.rzk.md:1590]: Binary operators are placed on the end of the preceding line, Although `,` isn't strictly treated as a binary operator in `rzk` I believe.

File: src/simplicial-hott/08-covariant.rzk.md (hunk 1 above)
Comment (fredrik-bakke) [reply src/simplicial-hott/08-covariant.rzk.md:1585]: The convention is to place spaces after opening parentheses on new lines. This is to neatly align indented code, and highlight the tree structure of definitions in `rzk`.

File: src/simplicial-hott/08-covariant.rzk.md (hunk 4 above)
Comment (emilyriehl) [reply src/simplicial-hott/08-covariant.rzk.md:1583]: Yeah let's go with "hom" instead of "representable" since it's shorter. :)

**Pull request merged**